content/en/agent/logs/advanced_log_collection.md

```diff
@@ -0,0 +1,252 @@
+
+{{< tabs >}}
+{{% tab "exclude_at_match" %}}
+
+| Parameter          | Description                                                                                        |
+|--------------------|----------------------------------------------------------------------------------------------------|
+| `exclude_at_match` | If the specified pattern is contained in the message, the log is excluded and not sent to Datadog. |
+
+For example, filtering out logs that contain a Datadog email address:
+
+```yaml
+logs:
+  - type: file
+    path: /my/test/file.log
+    service: cardpayment
+    source: java
+    log_processing_rules:
+    - type: exclude_at_match
+      name: exclude_datadoghq_users
+      ## Regexp can be anything
+      pattern: \w+@datadoghq.com
+```
+
+{{% /tab %}}
+{{% tab "include_at_match" %}}
+
+| Parameter          | Description                                                                       |
+|--------------------|-----------------------------------------------------------------------------------|
+| `include_at_match` | Only logs with a message that includes the specified pattern are sent to Datadog. |
+
+For example, sending only logs that contain a Datadog email address:
+
+```yaml
+logs:
+  - type: file
+    path: /my/test/file.log
+    service: cardpayment
+    source: java
+    log_processing_rules:
+    - type: include_at_match
+      name: include_datadoghq_users
+      ## Regexp can be anything
+      pattern: \w+@datadoghq.com
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## Scrub sensitive data in your logs
+
+If your logs contain sensitive information that you want to redact, configure the Datadog Agent to scrub sensitive sequences by using the `log_processing_rules` parameter in your configuration file with the **mask_sequences** `type`.
+
+This replaces all matched groups with `replace_placeholder` parameter value.
+
+For example, redact credit card numbers:
+
+```yaml
+logs:
+ - type: file
+   path: /my/test/file.log
+   service: cardpayment
+   source: java
+   log_processing_rules:
+      - type: mask_sequences
+        name: mask_credit_cards
+        replace_placeholder: "[masked_credit_card]"
+        ##One pattern that contains capture groups
+        pattern: (?:4[0-9]{12}(?:[0-9]{3})?|[25][1-7][0-9]{14}|6(?:011|5[0-9][0-9])[0-9]{12}|3[47][0-9]{13}|3(?:0[0-5]|[68][0-9])[0-9]{11}|(?:2131|1800|35\d{3})\d{11})
+```
+
+## Multi-line aggregation
+
+If your logs are not sent in JSON and you want to aggregate several lines into one single entry, configure the Datadog Agent to detect a new log using a specific regex pattern instead of having one log per line. This is accomplished by using the `log_processing_rules` parameter in your configuration file with the **multi_line** `type` which aggregates all lines into one single entry until the given pattern is detected again.
```

Comment:
```suggestion
If your logs are not sent in JSON and you want to aggregate several lines into a single entry, configure the Datadog Agent to detect a new log using a specific regex pattern instead of having one log per line. This is accomplished by using the `log_processing_rules` parameter in your configuration file with the **multi_line** `type` which aggregates all lines into one single entry until the given pattern is detected again.
```